interfaces/builtin/dsp_control.go

```diff
@@ -0,0 +1,111 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/interfaces/udev"
+)
+
+const dspControlSummary = `allows controlling digital signal processors on certain boards`
+
+const dspControlBaseDeclarationPlugs = `
+  dsp-control:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const dspControlBaseDeclarationSlots = `
+  dsp-control:
+    allow-installation:
+      slot-snap-type:
+        - core
```

Comment:
@anonymouse64  would it make sense instead to make it a hardware interface were the slot needs to be declated on the gadget with the flavor for the device? and then the plug needs to match?

Reply:
we could do it that way instead if you prefer, I'm not sure if it makes sense to have multiple flavor plugs on a device, it probably doesn't in which case having the slot come from the gadget makes sense

Reply:
Yes, I think it would make more sense because then as I said the gadget slot would declare what flavor it exposes, given that the device nodes made accessible would/could vary by flavor

Reply:
okay I will refactor this to be a gadget exposed slot with the flavor declared in the slot like the gpio/pwm slots. Do you think we should just make this interface something like `dsp` instead of `dsp-control` now?

Reply:
yes, dsp might be a better name if this is essentially a hardware interface, though the microcode loading might be too much for the plain name

Reply:
okay, I have refactored this PR (via force push) to be a gadget slotted interface with the flavor attribute on the slot side, and also removed the base declaration plug bits to match i.e. i2c, gpio, etc. and then also renamed to dsp